geopandas/tools/_show_versions.py

```diff
@@ -58,33 +58,33 @@ def _get_C_info():
             geos_dir = None
 
     try:
-        import fiona
+        import pyogrio
 
-        gdal_version = fiona.env.get_gdal_release_name()
+        gdal_version = pyogrio.__gdal_version_string__
+        gdal_dir = None
     except Exception:
         gdal_version = None
     try:
-        import fiona
+        # get_gdal_data_path is only available in pyogrio >= 0.4.2
+        from pyogrio import get_gdal_data_path
 
-        gdal_dir = fiona.env.GDALDataFinder().search()
+        gdal_dir = get_gdal_data_path()
     except Exception:
-        gdal_dir = None
+        pass
```

Comment:
In the latest version, you set the min version of pyogrio to 0.5, so `gdal_dir` should always be assignable in the try block above on line 64 (if pyogrio is available)